### PR TITLE
Fix failures to h2spec test and make h2spec test more resilient

### PR DESCRIPTION
**Pull request by shinrich**
Another solution to PR #6879 

Adjusts a change in #6732 to ensure that the error frame is emitted.

Also adjusts the timeout for h2spec from the default of 2 seconds to 10.  I was seeing failures in the concurrent stream overflow test (5.1.2) where sometimes it would succeed and sometimes timeout.  This was with and without the changes in #6732.

### CODE REVIEW

File: proxy/http2/Http2Stream.cc [hunk 1]
```diff
@@ -451,12 +451,9 @@ Http2Stream::initiating_close()
     }
     // Send EOS to let SM know that we aren't sticking around
     if (_sm && read_vio.cont) {
-      // Only bother with the EOS if we haven't sent the write complete
-      if (!sent_write_complete) {
```
Comment (masaori335) [inline proxy/http2/Http2Stream.cc:455]: Does this mean this function sends VC_EVENT_EOS even if VC_EVENT_WRITE_COMPLETE was sent? Looks a bit weird.

Comment (shinrich) [reply proxy/http2/Http2Stream.cc:455]: Agreed.  I was trying to back things out of the Http2 part of #6732.  Let me try again with only the h2spec timeout extension.

**Comment by masaori335** [comment]
[approve ci autest]

autest was skipped :(
https://ci.trafficserver.apache.org/job/autest-github/13481/console

**Comment by shinrich** [comment]
Closing.  Parts will reappear in a new PR.
